vendor/k8s.io/kubernetes/pkg/controller/statefulset/stateful_set.go

```diff
@@ -450,7 +450,11 @@ func (ssc *StatefulSetController) sync(key string) error {
 // syncStatefulSet syncs a tuple of (statefulset, []*v1.Pod).
 func (ssc *StatefulSetController) syncStatefulSet(set *apps.StatefulSet, pods []*v1.Pod) error {
 	glog.V(4).Infof("Syncing StatefulSet %v/%v with %d pods", set.Namespace, set.Name, len(pods))
-	if err := ssc.control.UpdateStatefulSet(set, pods); err != nil {
+	setCopy, err := api.Scheme.DeepCopy(set)
```

Comment:
yeah seems like decoded to internal version, I will open upstream pull and we can take it from there